servant/src/Servant/Utils/Links.hs

```diff
@@ -105,11 +105,7 @@ import Data.List
 import Data.Proxy ( Proxy(..) )
 import qualified Data.Text as Text
 import qualified Data.ByteString.Char8 as BSC
-#if !MIN_VERSION_base(4,8,0)
-import Data.Monoid ( Monoid(..), (<>) )
-#else
-import Data.Monoid ( (<>) )
-#endif
+import Data.Monoid.Compat ( Monoid(..), (<>) )
```

Comment:
I think if you import `Prelude.Compat` here, you don't need to import the `Monoid` type class.   That may be a more robust solution.  You still will need `(<>)` though...